bluemira/codes/process/_solver.py

```diff
@@ -141,7 +142,18 @@ def __init__(self, params: Union[Dict, ParameterFrame], build_config: BuildConfi
                 f"'{quoted_delim.join(_build_config.keys())}'."
             )
 
-    def execute(self, run_mode: RunMode) -> ParameterFrame:
+        # Get process defaults from template file
+        mm = _make_writer(self.template_in_dat).data
+        process_inputs = make_dataclass("ProcesInputs", list(mm.keys()))(
+            {key: value.get_value for key, value in mm.items()}
+        )
+
+        self.params = make_mapped_default_parameter_frame(
+            process_inputs, mappings, ProcessSolverParams
+        )
+        self.params.update_from_frame(params)
```

Comment:
this is also not great as we end up reading in the template twice. Here and in setup.py. Wanted to keep it separate from setup though...